app/models/iterations/project_iterations.rb

```diff
@@ -10,27 +10,49 @@ def current_iteration_start
 
     def past_iterations
       (0...length).map do |iteration_number|
-        start_date = start_date(iteration_number)
-        end_date = end_date(start_date)
-        PastIteration.new(start_date: start_date,
-                          end_date: end_date,
-                          project: @project,
-                          iteration_number: iteration_number + 1)
+        start_at = start_date(iteration_number)
+        end_at = end_date(start_at)
+
+        PastIteration.new(
+          start_date: start_at,
+          end_date: end_at,
+          stories: stories_from(start_at, end_at),
+          iteration_number: iteration_number + 1
+        )
       end
     end
 
     private
 
+    attr_reader :project
+
+    def stories_from(start_at, end_at)
+      stories.select do |story|
+        story.accepted_at >= start_at && story.accepted_at <= end_at
```

Comment:
Isn't it better to do these conditionals with a query instead of directly in Ruby side?

Reply:
Dunno, I moved from a query because it was doing a lot of queries to get all this...

Only if we have a query to bring grouped those stories by start/end_at...

Reply:
🤔

Reply:
@rodrigovirgilio ☝️